recognition/src/linemod.cpp

```diff
@@ -1662,6 +1663,112 @@ pcl::LINEMOD::detectTemplatesSemiScaleInvariant (
     }
   }
 }
+//////////////////////////////////////////////////////////////////////////////////////////////
+void
+pcl::LINEMOD::evaluateDetections(
+  const std::vector<QuantizableModality*>& modalities,
+  const std::vector<LINEMODDetection>& inputDetections,
+  const std::vector<SparseQuantizedMultiModTemplate>& inputTemplates,
+  std::vector<float>& evaluationScores
+) const
+{
+  // create energy maps
+  std::vector<EnergyMaps> modality_energy_maps;
+
+  const size_t nr_modalities = modalities.size();
```

Comment:
`modality_energy_maps` has known fixed size:

```suggestion
  const size_t nr_modalities = modalities.size();
  std::vector<EnergyMaps> modality_energy_maps;
  modality_energy_maps.reserve(nr_modalities);
  
```

Reply:
applied